### DISCUSSION TITLE
How to get the position or co-ordinates of the words in PDF ?

### DISCUSSION THREAD

**Discussion by Laxmi530** [jsvine/pdfplumber#708] [Ask for help with specific PDFs]
Hai, 
Thank you for providing such a beautiful library.
Trying to extract the bounding box or the position or the co-ordinates of the word in the form of [x, y, x+width, y+height].
Can someone please help me how can I do this.

**Comment by jsvine** [comment]
Hi @Laxmi530, and thanks for the kind words about the library. What have you tried doing so far? (Have you tried examining the values returned by `page.extract_words()`?)

**Reply by Laxmi530** [reply]
@jsvine Thanks for the replay. 
Tried that method but quite confusing it for me because it returns list of dictionary. From the dictionary x0+width=x1, this one I got but the confusion creates top, doctop, bottom. Now I got the two coordinates remaining two coordinates I can get once got the height of the word and that is confusing me which one is height out of these (top, doctop, bottom) three. 

`[{'text': 'sample', 'x0': 294.83999625, 'x1': 334.6118514454099, 'top': 124.19106199999999, 'doctop': 124.19106199999999, 'bottom': 134.151437, 'upright': True, 'direction': 1}, {'text': 'Page', 'x0': 476.16000375, 'x1': 506.41565264999997, 'top': 128.93640604999996, 'doctop': 128.93640604999996, 'bottom': 136.97670604999996, 'upright': True, 'direction': 1}]`

**Reply by jsvine** [reply]
Thanks for clarifying. I believe what you're looking for is `(x0, top, x1, bottom)`. You can find more details about those properties in the readme, e.g., [here](https://github.com/jsvine/pdfplumber#char-properties).

**Reply by Laxmi530** [reply]
@jsvine Thanks for the help.
I noticed one thing while extracting text from pdf if there is two or more words in pdf with space separated like `Hello World` in out put it returns `HelloWorld` like this. So is there any solution available.
How many language support this library?

**Reply by jsvine** [reply]
> I noticed one thing while extracting text from pdf if there is two or more words in pdf with space separated like Hello World in out put it returns HelloWorld like this. So is there any solution available.

Have you tried using the `x_tolerance` settings described in the documentation? If so, and that didn't work, could you share the PDF you're parsing and the code you've used? That will make it easier to answer your question more precisely.

>  How many language support this library?

If you mean programming languages: This is specifically a Python library; I don't think it has been ported to other programming languages, but other programming languages may have similar libraries. If you mean written languages: This library should be able to handle PDF documents written in any language, although it is currently primarily tested on those that are written left-to-right, top-to-bottom.

**Reply by Laxmi530** [reply]
@jsvine Thanks for the clarification.
Let me try `x_tolerance` and I will get back to you.


> If you mean written languages: This library should be able to handle PDF documents written in any language, although it is currently primarily tested on those that are written left-to-right, top-to-bottom.

Thanks for the clarification.
